base/uk.ac.stfc.isis.ibex.scriptgenerator/python_support/script_definition_loader.py

```diff
@@ -84,14 +93,41 @@ def getParameters(self) -> List[PythonActionParameter]:
 
         kwargs_with_defaults = []
 
+        enum_parameter_names = []
+        
+        try:
+            enum_parameter_names = getattr(self.script_definition,"enum_params")
+        except:
+            enum_parameter_names = []
+
+
         for arg in arguments:
             if arguments[arg].default == arguments[arg].empty:
                 action_parameter = PythonActionParameter(arg, arg, False)
             elif isinstance(arguments[arg].default, CopyPreviousRow):
                 # If none copy the previous row's value over
-                action_parameter = PythonActionParameter(arg, str(arguments[arg].default.value), True)
+                if arg in enum_parameter_names:
+
+                    enum = getattr(self.script_definition, arg+"_enum")
+
+                    enum_values = [str(item.name) for item in enum]
+                    action_parameter = PythonActionParameter(arg, enum_values[0], True, True, ListConverter().convert(enum_values, gateway._gateway_client))
+                else:
+                    try:
+                        action_parameter = PythonActionParameter(arg, str(arguments[arg].default.value), True)
+                    except:
+                        action_parameter = PythonActionParameter(arg, str(arguments[arg].default), True)
             else:
-                action_parameter = PythonActionParameter(arg, str(arguments[arg].default), False)
+                if arg in enum_parameter_names:
+                    enum = getattr(self.script_definition, arg+"_enum")
+                    enum_values = [str(item.name) for item in enum]
+                    action_parameter = PythonActionParameter(arg, enum_values[0], False, True, ListConverter().convert(enum_values, gateway._gateway_client))
+                else:
+                    try:
+                        action_parameter = PythonActionParameter(arg, str(arguments[arg].default.value), False)
+                    except:
+                        action_parameter = PythonActionParameter(arg, str(arguments[arg].default),False )
+            
```

Comment:
These two sections are similar enough that it may be worth pulling out a function here, it looks like it would just need two arguments, the default value or default, and true or false.

Reply:
done